drivers/apds99xx/apds99xx.c

```diff
@@ -0,0 +1,522 @@
+/*
+ * Copyright (C) 2018 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_apds99xx
+ * @brief       Device driver for the Broadcom APDS99XX proximity and ambient light sensor
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ * @file
+ * @{
+ */
+
+#include <string.h>
+#include <stdlib.h>
+
+#include "apds99xx_regs.h"
+#include "apds99xx.h"
+
+#include "irq.h"
+#include "log.h"
+#include "xtimer.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#if ENABLE_DEBUG
+
+#define ASSERT_PARAM(cond) \
+    if (!(cond)) { \
+        DEBUG("[apds99xx] %s: %s\n", \
+              __func__, "parameter condition (" # cond ") not fulfilled"); \
+        assert(cond); \
+    }
+
+#define DEBUG_DEV(f, d, ...) \
+        DEBUG("[apds99xx] %s i2c dev=%d addr=%02x: " f "\n", \
+              __func__, d->params.dev, APDS99XX_I2C_ADDRESS, ## __VA_ARGS__);
+
+#else /* ENABLE_DEBUG */
+
+#define ASSERT_PARAM(cond) assert(cond);
+#define DEBUG_DEV(f, d, ...)
+
+#endif /* ENABLE_DEBUG */
+
+#define ERROR_DEV(f, d, ...) \
+        LOG_ERROR("[apds99xx] %s i2c dev=%d addr=%02x: " f "\n", \
+                  __func__, d->params.dev, APDS99XX_I2C_ADDRESS, ## __VA_ARGS__);
+
+#define EXEC_RET(f) { \
+    int _r; \
```

Comment:
Thanks for your review.

> It would be better not to assign the value to _r inside of the if statement.

I'm wondering why? What is the reason.

> Also, is the intermediate variable really needed here and below?

Yes, if we want to print out the error in the debug message.